src/goto-programs/goto_contractor.cpp

```diff
@@ -0,0 +1,453 @@
+    h = create_function_from_expr2t(to_mul2t(expr).side_2);
+    f = new Function(*vars, (*g)(*vars) * (*h)(*vars));
+    break;
+  case expr2t::expr_ids::div_id:
+    g = create_function_from_expr2t(to_div2t(expr).side_1);
+    h = create_function_from_expr2t(to_div2t(expr).side_2);
+    f = new Function(*vars, (*g)(*vars) / (*h)(*vars));
+    break;
+  case expr2t::expr_ids::symbol_id:
+  {
+    int index = create_variable_from_expr2t(expr);
+    if(index != NOT_FOUND)
+      f = new Function(*vars, (*vars)[index]);
+    else
+      message_handler.error("ERROR: MAX VAR SIZE REACHED");
+    break;
+  }
+  case expr2t::expr_ids::typecast_id:
+    f = create_function_from_expr2t(to_typecast2t(expr).from);
+    break;
+  case expr2t::expr_ids::constant_int_id:
+  {
+    const ExprConstant &c =
+      ExprConstant::new_scalar(to_constant_int2t(expr).value.to_int64());
+    f = new Function(*vars, c);
+    break;
+  }
+  default:
+    f = nullptr;
+  }
+  //if(is_arith_expr(expr) && !is_modulus2t(expr) && !is_abs2t(expr))
+  /*
+  if(is_add2t(expr))
+  {
+    g = create_function_from_expr2t(to_add2t(expr).side_1);
+    h = create_function_from_expr2t(to_add2t(expr).side_2);
+    f = new Function(*vars, (*g)(*vars) + (*h)(*vars));
+  }
+  else if(is_sub2t(expr))
+  {
+    g = create_function_from_expr2t(to_sub2t(expr).side_1);
+    h = create_function_from_expr2t(to_sub2t(expr).side_2);
+    f = new Function(*vars, (*g)(*vars) - (*h)(*vars));
+  }
+  else if(is_mul2t(expr))
+  {
+    g = create_function_from_expr2t(to_mul2t(expr).side_1);
+    h = create_function_from_expr2t(to_mul2t(expr).side_2);
+    f = new Function(*vars, (*g)(*vars) * (*h)(*vars));
+  }
+  else if(is_div2t(expr))
+  {
+    g = create_function_from_expr2t(to_div2t(expr).side_1);
+    h = create_function_from_expr2t(to_div2t(expr).side_2);
+    f = new Function(*vars, (*g)(*vars) / (*h)(*vars));
+  }
+  else if(is_symbol2t(expr))
+  {
+    int index = create_variable_from_expr2t(expr);
+    if(index != -1)
+      f = new Function(*vars, (*vars)[index]);
+    else
+    {
+      message_handler.error("ERROR: MAX VAR SIZE REACHED");
+      //exit(1);
+    }
+  }
+  else if(is_typecast2t(expr))
+  {
+    f = create_function_from_expr2t(to_typecast2t(expr).from);
+  }
+  else if(is_constant_int2t(expr))
+  {
+    const ExprConstant &c =
+      ExprConstant::new_scalar(to_constant_int2t(expr).value.to_int64());
+    f = new Function(*vars, c);
+  }
+  else if(is_comp_expr(expr))
+  {
+    //Abort contractor
+  }*/
+  return f;
+}
+
+int goto_contractort::create_variable_from_expr2t(irep_container<expr2t> expr)
+{
+  std::string var_name = to_symbol2t(expr).get_symbol_name().c_str();
+  int index = map->find(var_name);
+  if(index == NOT_FOUND)
+  {
+    int new_index = map->add_var(var_name, to_symbol2t(expr));
+    if(new_index != NOT_FOUND)
+      return new_index;
+    return NOT_FOUND;
+  }
+  return index;
+}
+
+bool goto_contractort::run1()
```

Comment:
can you rename this function to something that's more meaningful?

Reply:
Done.